server/statistics/avg_over_time.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import (
+	"time"
+
+	"github.com/phf/go-queue/queue"
+)
+
+type deltaWithInterval struct {
+	delta    float64
+	interval time.Duration
+}
+
+// AvgOverTime maintains change rate in the last avgInterval.
+//
+// AvgOverTime takes changes with their own intervals,
+// stores recent changes that happened in the last avgInterval,
+// then calculates the change rate by (sum of changes) / (sum of intervals).
+type AvgOverTime struct {
+	que         *queue.Queue
+	deltaSum    float64
+	durSum      time.Duration
```

Comment:
how about `duration`? deltaSum and durSum have the same prefix and suffix

Reply:
I don't think it is a problem.

Reply:
They are similar and make it difficult to read

Reply:
Renamed to `intervalSum`.